CMakeLists.txt

```diff
@@ -474,6 +482,25 @@ if(WITH_CORE)
   find_package(EXIV2 REQUIRED)
 
   # optional
+  if (WITH_SFCGAL)
+
+    if(POLICY CMP0167) # avoid deprecation warning from CGAL
+      cmake_policy(SET CMP0167 NEW)
+    endif()
+
+    find_package(SFCGAL) # SFCGAL provider
+    message(STATUS "Found SFCGAL: ${SFCGAL_VERSION} ${SFCGAL_DIR}")
+    if(SFCGAL_VERSION VERSION_LESS "2.1")
+      message(FATAL_ERROR "Cannot build QGIS using SFCGAL older than 2.1")
+    endif()
+    if (WITH_SFCGAL)
+      add_definitions(-DWITH_SFCGAL)
```

Comment:
can you attach the definition to the `qgis_core` target instead making it global using `target_compile_definitions`?